aquarius/events/decryptor.py

```diff
@@ -26,7 +30,10 @@ def decrypt_ddo(w3, provider_url, contract_address, chain_id, txid):
     }
 
     response = requests.post(provider_url + "/api/services/decrypt", json=payload)
+
     if response.status_code == 201:
+        if sha256(response.text.encode("utf-8")).hexdigest() != hash.hex():
```

Comment:
this is already done somewhere in the processors, specifically to avoid passing this extra hash parameter. We are checking the correctness of decrypt. My reasoning is that this function should only do decrypt, not decrypt_and_check_hash.

Reply:
what is the use of hash ? Why it's there?
1. if data is unencrypted, the only possible use would be to use it as integrity check (since it's blockchain, that is not the case)  - no use
2. to protect ddo integrity when a 3rd party is involved. In out case, it's the provider. We need to make sure that what provider is returning is the same as publisher set.
Pottential vector attack if we don't use hash:
  - special crafted provider
  - Alice publishes a dataset using that provider
  - When aqua is caching ddo, provider will change ddo.service.datatokenAddress to a malicious address.   So the consumer will buy a bogus datatoken (provider will allow consume), and attacker gets the money. publisher gets nothing

So, the hash check must be done in the decrypt function, anywhere else it's a nonsense

Reply:
Thank you for explaining. I appreciate it.